src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```diff
@@ -0,0 +1,202 @@
+        return HelpFile.FETCHER_MEDLINE;
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        String cleanQuery = identifier.trim().replace(';', ',');
+
+        List<BibEntry> entry = fetchMedline(cleanQuery);
+
+        if (entry.isEmpty()) {
+            LOGGER.warn(Localization.lang("No references found"));
+        } else if (entry.size() > 1) {
+            LOGGER.info("Fetcher " + getName() + " " +
+                    "found more than one result for identifier " + identifier
+                    + ". We will use the first entry.");
+        }
+        return Optional.of(entry.get(0));
+    }
+
+    @Override
+    public List<BibEntry> performSearch(String query) throws FetcherException {
+        final int NUMBER_TO_FETCH = 50;
+        List<BibEntry> entryList = new LinkedList<>();
+
+        if (!query.isEmpty()) {
+            String searchTerm = replaceCommaWithAND(query);
+
+            // get the ids from entrez
+            getIds(searchTerm, 0, 1);
+
+            if (count == 0) {
+                LOGGER.warn(Localization.lang("No references found"));
+                return Collections.emptyList();
+            }
+
+            getIds(searchTerm, 0, NUMBER_TO_FETCH-1);
+
+            List<BibEntry> bibs = fetchMedline(ids);
+            entryList.addAll(bibs);
+
+            return entryList;
+        }
+        throw new FetcherException(Localization.lang("Input error") + ". " + Localization.lang("Please enter a comma separated list of Medline IDs (numbers) or search terms."));
+    }
+
+    static URL createSearchUrl(String term, int start, int pacing) throws URISyntaxException, MalformedURLException {
+        term = replaceCommaWithAND(term);
+        URIBuilder uriBuilder = new URIBuilder(API_MEDLINE_SEARCH);
+        uriBuilder.addParameter("db", "pubmed");
+        uriBuilder.addParameter("retmax", Integer.toString(pacing));
+        uriBuilder.addParameter("retstart", Integer.toString(start));
+        uriBuilder.addParameter("sort", "relevance");
+        uriBuilder.addParameter("term", term);
+        return uriBuilder.build().toURL();
+    }
+
+    /**
+     * @see <a href="https://www.ncbi.nlm.nih.gov/books/NBK25499/table/chapter4.T._valid_values_of__retmode_and/?report=objectonly">www.ncbi.nlm.nih.gov</a>
+     */
+    static URL createFetchUrl(String id) throws URISyntaxException, MalformedURLException {
+        URIBuilder uriBuilder = new URIBuilder(API_MEDLINE_FETCH);
+        uriBuilder.addParameter("db", "pubmed");
+        uriBuilder.addParameter("retmode", "xml");
+        uriBuilder.addParameter("id", id);
+        return uriBuilder.build().toURL();
+    }
+
+    /**
+     * Fetch and parse an medline item from eutils.ncbi.nlm.nih.gov.
+     *
+     * @param id One or several ids, separated by ","
+     * @return Will return an empty list on error.
+     */
+    private static List<BibEntry> fetchMedline(String id) {
```

Comment:
It appears to me that the id-based part can be refactored to use the `Parser`-helper (I think the class was called IdBasedParserFetcher). For this MedlineImporter needs to implement the Parser interface but this should be easy.